dsa-solutions/gfg-solutions/Easy problems/check-for-balanced-tree.md

```diff
@@ -0,0 +1,255 @@
+---
+id: check-for-balanced-tree
+title: Check for balanced Tree
+sidebar_label: 4 Check for balanced tree
+tags:
+  - Binary Tree
+  - Depth-First Search
+  - Recursion
+  - LeetCode
+  - C++
+  - Java
+  - Python
+description: "This is a solution to the Balanced Binary Tree problem on LeetCode."
+---
+
+## Problem Description
+
+Given a binary tree, determine if it is height-balanced.
+
+A height-balanced binary tree is defined as:
+
+- A binary tree in which the left and right subtrees of every node differ in height by no more than 1.
+
+### Examples
+
+**Example 1:**
+
+![input Tree](https://assets.leetcode.com/uploads/2020/10/06/balance_1.jpg)
+
+```
+Input: root = [3,9,20,null,null,15,7]
+Output: true
+```
+
+**Example 2:**
+
+![input Tree](https://assets.leetcode.com/uploads/2020/10/06/balance_2.jpg)
+```
+Input: root = [1,2,2,3,3,null,null,4,4]
+Output: false
+```
+
+**Example 3:**
+
+```
+Input: root = []
+Output: true
+```
+
+### Constraints
+
+- The number of nodes in the tree is in the range `[0, 5000]`.
+- $ -10^4 <=$ Node.val $<= 10^4 $
```

Comment:
replace `- $ -10^4 <=$ Node.val $<= 10^4 $` to `- $-10^4 <=$ Node.val $<= 10^4$`